pkg/services/ngalert/api/tooling/definitions/alertmanager.go

```diff
@@ -484,10 +485,26 @@ func (c *Config) UnmarshalJSON(b []byte) error {
 		}
 	}
 
-	if c.Route != nil {
-		if err := c.Route.UnmarshalYAML(noopUnmarshal); err != nil {
-			return err
-		}
+	if c.Route == nil {
+		return fmt.Errorf("no routes provided")
+	}
+
+	// Route is a recursive structure that includes validation in the yaml unmarshaler.
+	// Therefore, we'll redirect json -> yaml to utilize these.
+	b, err := yaml.Marshal(c.Route)
+	if err != nil {
+		return errors.Wrap(err, "marshaling route to yaml for validation")
+	}
+	err = yaml.Unmarshal(b, c.Route)
+	if err != nil {
+		return errors.Wrap(err, "unmarshaling route for validations")
+	}
+
+	if len(c.Route.Receiver) == 0 {
+		return fmt.Errorf("root route must specify a default receiver")
+	}
+	if len(c.Route.Match) > 0 || len(c.Route.MatchRE) > 0 {
```

Comment:
We had an internal discussion that these fields will be deprecated and better use the `Matchers` instead. Therefore I think we would like to validate also this field.

Reply:
I copied this directly from the upstream code here: https://github.com/prometheus/alertmanager/blob/master/config/config.go#L453-L466

I was also surprised it didn't include the new fields, but didn't want to deviate.